.buildkite/pipeline.yml

```diff
@@ -1,26 +1,28 @@
-common-params:
-  &docker-container
-  docker#v3.8.0:
-    image: "public.ecr.aws/automattic/android-build-image:9d1710c7d74d1349d84e94465fe7b3a90216e92a"
-    propagate-environment: true
-    environment:
-      # DO NOT MANUALLY SET THESE VALUES!
-      # They are passed from the Buildkite agent to the Docker container
-      - "AWS_ACCESS_KEY"
-      - "AWS_SECRET_KEY"
+agents: { queue: "android" }
+
+x-common-params:
+  - &ci_toolkit
+      automattic/a8c-ci-toolkit#3.0.1
+  - &nvm_plugin
+    automattic/nvm#0.3.0:
+      # This is an attempt to fix curl error (92) during installing nodejs.
+      curlrc: --http1.1
 
 steps:
   - label: "Publish to Maven Local (to verify build)"
     command: |
+      sdkmanager --install "ndk;23.1.7779620"
```

Comment:
I think your suggestion @AliSoftware will work, although there's an issue with permissions now. CI fails with `Warning: Failed to read or create install properties file.` and it led me to this [SO answer](https://stackoverflow.com/questions/64464864/sdkmanager-warning-failed-to-read-or-create-install-properties-file) - what would you say is the best way to address? I guess calling `sudo` here is not an option.

Reply:
Mmmh damn. Not sure tbh.

I'd not be that surprised if that was in the context of a Docker container installing stuff on the host (via shared volume) because we know inside Docker the user is usually mapped to user 0 = root by default.
But I'm surprised that the android SDK would be installed as `root` in the `android` agent when this code runs here. Did you check (e.g. `ls -l` to debug) that Unix permissions/ownership was the root cause (no pun intended 😛) of the installation failure, and not something else?

Reply:
Ah, right, that makes sense.

And what confused me at first is that I was sure that we used `sdkmanager` to install stuff in other cases / apps / repos in our command scripts… but maybe I confused it with `avdmanager` (which I think doesn't install AVDs in `/usr/local/android`?)… or maybe the packages we install with `sdkmanager` in those cases I was thinking of… are different type of packages, that are installed elsewhere? 🤔  Or maybe what I was thinking about was for Appium tests in Android, which use a different agent (ad runs on Mac machines, not Docker nor AWS AMIs), so context is different…

Anyway, in the end not sure if there's a better solution than to add the NDK to the agents [at provision time in the Ansible Playbook](https://github.com/Automattic/buildkite-ci/blob/trunk/src/agents/android/android-playbook.yml#L31-L38) and redeploy them. And sure, if we later need a newer version of the NDK we'll have to update the agent and redeploy once again… but I guess it's similar to how we handle the JDK in a way, right?

Reply:
Thanks for the link to Ansible Playbook! I was looking for something like this. Yes, that might be the case - although NDK is really needed only here, so I'd prefer to install it here only. But if there's no other way, I'll prepare a PR for Ansible Playbook.

Reply:
Cc @crazytonyli @jkmassel in case you think of a way to only install it locally by the pipeline (and working around the unix permissions issues of being different users) as opposed to installing it at agent deployment time for everyone like I suggested above?

Reply:
In terms user/group of the `buildkite-agent` and `ec2-user` users, _I think_ they are suppose to be in the same group.

However, I think [this task](https://github.com/Automattic/buildkite-ci/blob/9e85e51e049f64b749b4675b09439d33528436d0/src/agents/android/android-playbook-tasks/setup-buildkite-agent.yml#L2-L6) is configured wrong: It's adding `ec2-user` to the `buildkite-agent` group, instead of the other way around.

```yaml
- name: Add Buildkite user to ec2-user group
  user:
    name: 'ec2-user'
    groups: buildkite-agent
    append: true
```

Reply:
btw I've discussed this with @jkmassel on 1:1 and it'd be probably best to add NDK installation to Ansible.